File: ncarg2d/src/libncarg_gks/awi/gscr.f
C
C	$Id: gscr.f,v 1.4 2000-07-12 16:39:56 haley Exp $
C                                                                      
C                Copyright (C)  2000
C        University Corporation for Atmospheric Research
C                All Rights Reserved
C
C This file is free software; you can redistribute it and/or modify
C it under the terms of the GNU Lesser General Public License as
C published by the Free Software Foundation; either version 2.1 of the
C License, or (at your option) any later version.
C
C This software is distributed in the hope that it will be useful, but
C WITHOUT ANY WARRANTY; without even the implied warranty of
C MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
C Lesser General Public License for more details.
C
C You should have received a copy of the GNU Lesser General Public
C License along with this software; if not, write to the Free Software
C Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
C USA.
C
      SUBROUTINE GSCR (WKID,CI,CR,CG,CB)
C
C  SET COLOUR REPRESENTATION
C
      INTEGER WKID,CI,ESCR
      REAL CR,CG,CB
      PARAMETER (ESCR=48)
C
      include 'gkscom.h'
C
C  Check if GKS is in the proper state.
C
      CALL GZCKST(7,ESCR,IER)
      IF (IER .NE. 0) RETURN
C
C  Check if workstation ID is valid.
C
      CALL GZCKWK(20,ESCR,WKID,IDUM,IER)
      IF (IER .NE. 0) RETURN
C
C  Check for invalid workstation categories.
C
      CALL GQWKC(WKID,IER,ICON,ITYP)
      CALL GQWKCA(ITYP,IER,ICAT)
      IF (ICAT .EQ. GMI) THEN
        ERS = 1
        CALL GERHND(33,ESCR,ERF)
        ERS = 0
        RETURN
      ELSE IF (ICAT .EQ. GINPUT) THEN
        ERS = 1
        CALL GERHND(35,ESCR,ERF)
        ERS = 0
        RETURN
      ELSE IF (ICAT .EQ. GWISS) THEN
        ERS = 1
        CALL GERHND(36,ESCR,ERF)
        ERS = 0
        RETURN
      ENDIF
C
C  Check if the workstation is currently open.
C
      CALL GZCKWK(25,ESCR,WKID,IDUM,IER)
      IF (IER .NE. 0) RETURN
C
C  Check if color index is non-negative.
C
      IF (CI .LT. 0) THEN
        ERS = 1
        CALL GERHND(92,ESCR,ERF)
        ERS = 0
        RETURN
      ENDIF
C
C  Check if colors are in range.
C
      IF (CR.LT.0. .OR. CR.GT.1. .OR. CG.LT.0. .OR.
     +    CG.GT.1. .OR. CB.LT.0. .OR. CB.GT.1.) THEN
        ERS = 1
        CALL GERHND(96,ESCR,ERF)
        ERS = 0
        RETURN
      ENDIF
C
C  Set function code and put out WKID, CI, and colors.
C
C  Set the flag CUFLAG to indicate that the interface call should go
C  only to the specifically designated workstation.
C
      CUFLAG = WKID
      FCODE = 56
      CONT  = 0
      CALL GZROI(0)
      IL1 = 2
      IL2 = 2
      ID(1) = WKID
      ID(2) = NOPICT
      IC1 = 1
      IC2 = 1
      IC(1) = CI
      RL1 = 3
      RL2 = 3
      RX(1) = CR
      RX(2) = CG
      RX(3) = CB
      CALL GZTOWK
      CUFLAG = -1
      IF (RERR .NE. 0) THEN
        ERS = 1
        CALL GERHND(RERR,ESCR,ERF)
        ERS = 0
      ENDIF
C
      RETURN
      END

File: ncarg2d/man/gks/gscr.m
.\"
.\"	$Id: gscr.m,v 1.11 2003-05-25 17:16:51 haley Exp $
.\"
.TH GSCR 3NCARG "March 1993" UNIX "NCAR GRAPHICS"
.SH NAME
GSCR (Set color representation) - associates a color value with a
color index.
.SH SYNOPSIS
CALL GSCR (WKID, CI, CR, CG, CB)
.SH C-BINDING SYNOPSIS
#include <ncarg/gks.h>
.sp
void gset_colr_rep(Gint ws_id, Gint colr_ind, const Gcolr_rep *colr_rep);
.SH DESCRIPTION
.IP WKID 12
(Integer, Input) - A workstation identifier.
WKID must be the same as that used in some previous GOPWK call.
.IP CI 12
(Integer, Input) - A color index.
.IP CR 12
(Real, Input) - An intensity value for red between 0. and 1. inclusive.
.IP CG 12
(Real, Input) - An intensity value for green between 0. and 1. inclusive.
.IP CB 12
(Real, Input) - An intensity value for blue between 0. and 1. inclusive.
.SH USAGE
Color in GKS is "indexed", i.e. color attributes are
assigned to primitives by using a color index.  The GKS
function GSCR is used to associate a color value with
color indices. 
.sp
It is  recommended that all color indices used in 
a given job be defined prior to calling any output primitive.
Since the result of a dynamic color change can be ambiguous, it is also
recommended that GSCR not be called to change a color value
after the original definitions.
.sp
GSCR may be called after opening the workstation for which
the color indices are to be defined, and not before that
workstation is opened.  Different color tables can be maintained
for different workstations.  For example, color index 2 for a
CGM file may be associated with "red" while at the same time it
may be associated with "green" for some X workstation.
.sp
It is important to stress that color index 0 defines the background
color. If any color indices are defined, then you 
should define the background color index 0. Otherwise 
you run the risk of having a user-defined color match 
the default background color.
.sp
For all GKS output primitives, color is assigned by an 
indexing scheme. The indices run from 0 to 
255, where 0 is the background color index and 1 is 
the foreground color index.  
.SH ACCESS
To use GKS routines, load the NCAR GKS-0A library 
ncarg_gks.
.SH SEE ALSO
Online: 
gqcr, gsplci, gspmci, gsfaci, gstxci, gqplci, gqpmci, gqfaci, gqtxci, gset_colr_rep
.sp
Hardcopy: 
User's Guide for NCAR GKS-0A Graphics;
NCAR Graphics Fundamentals, UNIX Version
.SH COPYRIGHT
Copyright (C) 1987-2003
.br
University Corporation for Atmospheric Research
.br

This documentation is free software; you can redistribute it and/or
modify it under the terms of the GNU General Public License as published
by the Free Software Foundation; either version 2 of the License, or
(at your option) any later version.

This software is distributed in the hope that it will be useful, but
WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
General Public License for more details.

You should have received a copy of the GNU General Public License
along with this software; if not, write to the Free Software
Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
USA.
